oci/layout/oci_dest.go

```diff
@@ -184,23 +193,20 @@ func (d *ociImageDestination) PutManifest(m []byte) error {
 		Architecture: runtime.GOARCH,
 		OS:           runtime.GOOS,
 	}
-	d.index.Manifests = append(d.index.Manifests, desc)
+	d.addManifest(&desc)
 
 	return nil
 }
 
-func ensureDirectoryExists(path string) error {
-	if _, err := os.Stat(path); err != nil && os.IsNotExist(err) {
-		if err := os.MkdirAll(path, 0755); err != nil {
-			return err
+func (d *ociImageDestination) addManifest(desc *imgspecv1.Descriptor){
+	for i, manifest := range d.index.Manifests {
+		if manifest.Digest.String() == desc.Digest.String() {
```

Comment:
> The image name should not be part of the reference.

But how do I then reference multiple images within the same OCI directory? I might be missing something here. Or are you suggesting that I create my own custom tag (eg combining image name with a version)?

> doesn't make sense to include the path to the image, because you can move the image 

This might be a misunderstanding. With path I mean the directory of where the OCI index lives. Then in order to push multiple images into the same index, I need to reference these images. Looking at ParseReference in oci_transport.go, tag seems to be used as the version of the image. It does not contain any name component.

So how do you suggest I use the current current _ociReference_ to store multiple images into the same OCI directory?